cpp/src/lance/format/schema.cc

```diff
@@ -378,6 +384,16 @@ ::arrow::Status CopyField(std::shared_ptr<Field> new_field,
   if (comp_idx >= components.size() || !new_field || !field) {
     return ::arrow::Status::OK();
   }
+
+  /// If this is a list<struct> node, we push the copy field into the child / struct node.
+  if (field->logical_type() == "list.struct") {
+    assert(field->children_.size() == 1);
+    if (new_field->children_.empty()) {
+      new_field->children_.emplace_back(field->children_[0]->Copy(false));
```

Comment:
This why Field.Get can no longer be `const`?

Reply:
Oh, no. Removing `const` is just a `clang-tidy` complaint. Because it returns a new object, no need to add `const` to it.